cloud-gcp/src/main/java/com/sequenceiq/cloudbreak/cloud/gcp/compute/GcpInstanceResourceBuilder.java

```diff
@@ -230,26 +232,43 @@ private void assignToExistingInstanceGroup(GcpContext context,
         String projectId = context.getProjectId();
         String zone = context.getLocation().getAvailabilityZone().value();
 
-        List<CloudResource> instanceGroupResources = filterGroupFromName(filterResourcesByType(context.getGroupResources(group.getName()),
-                ResourceType.GCP_INSTANCE_GROUP), group.getName());
-
-        if (!instanceGroupResources.isEmpty() && doesGcpInstanceGroupExist(compute, projectId, zone, instanceGroupResources.get(0))) {
-            LOGGER.info("adding instance {} to group {} in project {}", instance.getName(), group.getName(), projectId);
-            InstanceGroupsAddInstancesRequest request = createAddInstancesRequest(instance, projectId, zone);
-            AddInstances addInstances = compute.instanceGroups().addInstances(projectId,
-                    zone, instanceGroupResources.get(0).getName(), request);
-            try {
-                Operation execute = addInstances.execute();
-                verifyOperation(execute, buildableResource);
-            } catch (GoogleJsonResponseException e) {
-                LOGGER.error("Error in Google response, unable to add instance {} to group {} : {} for {}",
-                        instance.getName(), group.getName(), e.getMessage(), e.getDetails().getMessage());
-                throw new GcpResourceException(checkException(e), resourceType(), buildableResource.get(0).getName());
-            }
-        } else {
+        List<CloudResource> instanceGroups = filterInstanceGroupsByName(context.getGroupResources(group.getName()), group.getName());
+
+        if (instanceGroups.isEmpty()) {
+            LOGGER.info("skipping group assignment {} was not found in project {}", group.getName(), projectId);
+            return;
+        }
+        if (instanceGroups.size() > 1) {
+            LOGGER.warn("Found multiple instance groups to assign, trying to find an active one. Group {}, project {}", group.getName(), projectId);
+        }
+        CloudResource targetInstanceGroup = findExistingGroup(compute, projectId, zone, instanceGroups);
```

Comment:
Optional would be more elegant than using plain `null`

Reply:
I disagree but changed accordingly